"The reference assemblies for .NETFramework,Version=v4.0 were not found" appeared in Python project.

**Describe the bug**

![image](https://user-images.githubusercontent.com/82852072/136506667-0a111b73-4f2e-4a70-9e04-3c5788bffe92.png)

**Steps to Reproduce**

Create a Python Application project.

**Expected behavior**

It should be created successfully without any errors.

**Additional context and screenshots**

An error appears in the error list.

![image](https://user-images.githubusercontent.com/82852072/136506915-c05be372-21c8-4de1-b7c3-224d1a8000e1.png)

**Installation details**

![image](https://user-images.githubusercontent.com/82852072/136507123-f2d70fca-feeb-4401-87d6-9cafc610d3d8.png)

OK I put a zip here with them.
[pyprojfiles.zip](https://github.com/microsoft/PTVS/files/7672043/pyprojfiles.zip)
doesntworkatall.pyproj is the old one I am trying to get working.  Get that error when hitting run or build etc.
runsbutrebuilderrors.pyproj is the new one.  And when verifying things before uploading for you, I noticed that it runs fine and builds fine.  But if I do a rebuild on it it fails with the error:
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets(1217,5): error MSB3644: The reference assemblies for .NETFramework,Version=v4.0 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks
Done building project "DjangoWebProject1.pyproj" -- FAILED.

--- a/Python/Product/BuildTasks/Microsoft.PythonTools.targets
+++ b/Python/Product/BuildTasks/Microsoft.PythonTools.targets
@@ -24,7 +24,7 @@ permissions and limitations under the License.
     <_PTVSSupportPath Condition="'$(_PTVSSupportPath)' == ''">$(MSBuildThisFileDirectory)</_PTVSSupportPath>
     <_PTVSSupportPath Condition="!HasTrailingSlash('$(_PTVSSupportPath)')">$(_PTVSSupportPath)\</_PTVSSupportPath>
     <_PTVSBuildTasks Condition="'$(_PTVSBuildTasks)' == ''">$([MSBuild]::Unescape($(_PythonToolsPath)Microsoft.PythonTools.BuildTasks.dll))</_PTVSBuildTasks>
-    <_PTVSBuildTasks Condition="!Exists($(_PTVSBuildTasks))">$([MSBuild]::Unescape($(_PTVSSupportPath)Microsoft.PythonTools.BuildTasks.Core.dll))</_PTVSBuildTasks>  
+    <_PTVSBuildTasks Condition="!Exists($(_PTVSBuildTasks))">$([MSBuild]::Unescape($(_PTVSSupportPath)Microsoft.PythonTools.BuildTasks.Core.dll))</_PTVSBuildTasks>
   </PropertyGroup>
 
   <!-- 

--- a/Python/Product/BuildTasks/Microsoft.PythonTools.targets
+++ b/Python/Product/BuildTasks/Microsoft.PythonTools.targets
@@ -24,7 +24,18 @@ permissions and limitations under the License.
     <_PTVSSupportPath Condition="'$(_PTVSSupportPath)' == ''">$(MSBuildThisFileDirectory)</_PTVSSupportPath>
     <_PTVSSupportPath Condition="!HasTrailingSlash('$(_PTVSSupportPath)')">$(_PTVSSupportPath)\</_PTVSSupportPath>
     <_PTVSBuildTasks Condition="'$(_PTVSBuildTasks)' == ''">$([MSBuild]::Unescape($(_PythonToolsPath)Microsoft.PythonTools.BuildTasks.dll))</_PTVSBuildTasks>
-    <_PTVSBuildTasks Condition="!Exists($(_PTVSBuildTasks))">$([MSBuild]::Unescape($(_PTVSSupportPath)Microsoft.PythonTools.BuildTasks.Core.dll))</_PTVSBuildTasks>
+    <_PTVSBuildTasks Condition="!Exists($(_PTVSBuildTasks))">$([MSBuild]::Unescape($(_PTVSSupportPath)Microsoft.PythonTools.BuildTasks.Core.dll))</_PTVSBuildTasks>  
+  </PropertyGroup>
+
+  <!-- 
+    The global default value for target framework version seems to be v4.0.
+    This is used in various places, but specifically used by the GetReferenceAssemblyPaths target in Microsoft.Common.CurrentVersion.Targets.
+    The default version works with VS 2019 and lower, but it is not supported by VS 2022 and higher. This is causing pyproj files to break in VS2022.
+    Therefore, we're overriding it to 4.7.2, which is supported.
+  -->
+  <PropertyGroup>
+    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
+    <TargetFrameworkMoniker>.NETFramework,Version=$(TargetFrameworkVersion)</TargetFrameworkMoniker>
   </PropertyGroup>
 
   <!-- Creates a command object that will be displayed in the IDE.

